### DISCUSSION TITLE
How to make custom separator/lines and Termux (battery/wifi) help needed.

### DISCUSSION THREAD

**Discussion by Dan1jel** [fastfetch-cli/fastfetch#419] [Q&A]
How can I make lines like these, i must be really bad at finding stuff because I can't find any documentation about this. When I try to make a line in a specific length, it just get cut down.

Do someone know how to make this or what is needed for it to work? (Picture of demo below)

https://github.com/LinusDierheimer/fastfetch/blob/dev/screenshots/example2.png

I also wonder how i can Termux Battery and Wifi to work.

Tried this
```
--battery-format termux-battery-status | grep "percentage" | awk -F: '{ print $2}' | sed 's/.$//
```
And 

```
--wifi-key   "$(termux-wifi-connectioninfo | awk 'FNR == 9 { print $2}' | cut -d '"' -f2)"
```
Without any reaction except my command in the fastfetch output.

Any help is appreciated 🙏

**Comment by CarterLi** [comment]
> Do someone know how to make this or what is needed for it to work? (Picture of demo below)

It's https://github.com/LinusDierheimer/fastfetch/blob/dev/presets/examples/2

Load it with `--load-config presets/examples/2`

> I also wonder how i can Termux Battery and Wifi to work.

It's implemented inside fastfetch, don't hack it.

Please check if `~/../usr/libexec/termux-api` exists. [fastfetch calls it directly](https://github.com/LinusDierheimer/fastfetch/blob/dev/src/detection/battery/battery_android.c#L13) instead of its wrapper `termux-battery-status`

**Reply by Dan1jel** [reply]
Thanks for the reply. I have no intention of "hack" it, i just want to do my own theme. And right now i have no knowlage or documentation that helps me with that, if this is not possible anymore then i just have to accept that, but if not, is there a way to add custom lines/spacers?

**Reply by Dan1jel** [reply]
When I try the full length of the path that you provided, then it works, but just to type "termux-api BatteryStatus" do not, please see attached photo.

![Screenshot_20230218-204408.png](https://user-images.githubusercontent.com/4266424/219885212-bbdce13f-3f29-4d9c-92c8-dd41356a8352.png)

**Comment by CarterLi** [comment]
Expected result: https://github.com/LinusDierheimer/fastfetch/wiki#android-without-logo

**Comment by CarterLi** [comment]
Then everything is correct. What does `fastfetch -s battery:wifi --show-errors` print?

**Reply by Dan1jel** [reply]
![Screenshot_20230219-074026.png](https://user-images.githubusercontent.com/4266424/219933142-cfca77fd-0ef6-448b-90f7-068f11f52abd.png)

Think a picture says more, but let me know if you want me to copy and paste the code instead in future replies.

**Reply by CarterLi** [reply]
How about `env NO_CONFIG=1 fastfetch -s battery:wifi --show-errors`

**Reply by Dan1jel** [reply]
Same:
![Screenshot_20230219-090458.png](https://user-images.githubusercontent.com/4266424/219936392-e47e1782-4a04-4c52-887f-d78bd48daec0.png)

**Reply by CarterLi** [reply]
Okey, the support for NO_CONFIG is not released yet.

Please disable your user config (rename your `config.conf` to something else) and try again

**Reply by Dan1jel** [reply]
Same still... Before (2 days ago), i tried to uninstall it and install again with no difference. 

![Screenshot_20230219-092839.png](https://user-images.githubusercontent.com/4266424/219937318-04062c23-d35b-43ad-9ed2-1946540816e0.png)

![Screenshot_20230219-093600.png](https://user-images.githubusercontent.com/4266424/219937789-d26ceacd-3978-49a5-8600-6b6641e04c65.png)

**Reply by CarterLi** [reply]
I can confirm the same behavior with package installed by apt, but I can't reproduce it when building from source. Still don't know what's going wrong.
Can you try building fastfetch from source?

```sh
$ pkg i clang cmake
$ git clone https://github.com/LinusDierheimer/fastfetch.git
$ cd fastfetch
$ mkdir build && cd build
$ cmake ..
$ cmake --build . --target fastfetch
$ ./fastfetch -s battery:wifi
```

**Reply by Dan1jel** [reply]
You were correct, apt is having some issues, but build from source works alright. 

![Screenshot_20230219-104211.png](https://user-images.githubusercontent.com/4266424/219940504-12b7e2fc-0b2b-402a-8859-657b48c76dab.png)

**Reply by Dan1jel** [reply]
Thank you for helping me solve this issue, really appreciate you taking your time.

**Reply by CarterLi** [reply]
This issue has been fixed in v1.10.2

**Comment by Dan1jel** [comment]
> > Do someone know how to make this or what is needed for it to work? (Picture of demo below)
> 
> It's https://github.com/LinusDierheimer/fastfetch/blob/dev/presets/examples/2
> 
> Load it with `--load-config presets/examples/2`

I tried this and still get error, dose this only workwork with --kitten option? 

![Screenshot_20230219-151727.png](https://user-images.githubusercontent.com/4266424/219953972-dde489a8-0a1a-4d7d-92eb-7bf7f08c894f.png)

**Reply by CarterLi** [reply]
See https://github.com/termux/termux-app/pull/2973

**Reply by Dan1jel** [reply]
Could you please elaborate what i should look for in this?

In neofetch, i have line like the preview dose, and it works, but when assigning the key for the line, i get error because there is no option like that in fastfetch. 

I even tried to use the demo theme 2 and it got error, so what setting/option do i miss in my config?

**Reply by CarterLi** [reply]
![Screenshot_2023-02-20-01-24-51-497_com.termux.jpg](https://user-images.githubusercontent.com/6134068/219964301-c877c3ad-94d4-480d-90d2-021a4e0de7f5.jpg)

**Reply by CarterLi** [reply]
[`--set-keyless`](https://github.com/LinusDierheimer/fastfetch/blob/dev/CHANGELOG.md#164) was introduced in v1.6.4 and we are now in v1.9.1+

**Reply by Dan1jel** [reply]
My config looks like this and i still get the error, but your example works when I try it, must be something that is missing for the error to show?

```
#--set-keyless HardwareStart=┌───────── \e[1mHardware Information\e[0m ─────────┐
#--set-keyless SoftwareStart=├───────── \e[1mSoftware Information\e[0m ─────────┤
#--set-keyless InformationEnd=└────────────────────────────────────────┘
#--set-keyless Colors=    \e[38m \e[37m \e[36m \e[35m \e[34m \e[33m \e[32m \e[31m

--structure Break:Separator:HardwareStart:OS:Host:Kernel:Uptime:Packages:Shell:Display:DE:WM:WMTheme:Information:Theme:Icons:Font:Terminal:TerminalFont:CPU:GPU:Memory:Disk:Player:Media:LocalIP:Separator:InformationEnd:

--logo-padding-top 1
--color-keys 0
--separator "  "
--public-ip-url "ifconfig.me"
--weather-output-format "%t+-+%C+(%l)"

#--player-name spotify
--disk-folders /:/server/
--os-key 
--host-key 
#--chassis-key Chassis
--kernel-key 
--uptime-key 
#--processes-key Processes
--packages-key 
--shell-key 
#--display-key Display {1}
#--brightness-key Brightness ({1})
#--de-key DE
#--wm-key WM
#--wm-theme-key WM Theme
#--theme-key Theme
#--icons-key Icons
#--font-key Font
#--cursor-key Cursor
--terminal-key 
--terminal-font-key 
--cpu-key 
#--cpu-usage-key CPU Usage
--gpu-key 
--memory-key 
#--swap-key Swap
--disk-key 
--battery-key 
#--poweradapter-key Power Adapter {1}
#--locale-key Locale
--local-ip-key 
--public-ip-key 
--wifi-key 
#--weather-key Weather
--player-key 
--media-key 
#--datetime-key Date Time
#--vulkan-key Vulkan
#--opengl-key OpenGL
#--opencl-key OpenCL
#--users-key Users
```

Edit: updated code to show correctly, im aware of the # begining of the code, but cant use Fastfetch with the error code so thats why i put # infront of the affected options in the top

**Reply by CarterLi** [reply]
@LinusDierheimer Can you explain this issue?

**Reply by Dan1jel** [reply]
A little update @CarterLi & @LinusDierheimer .

I got it working, when i create a new file for ex 
`config.conf` i could not add the lines what do ever, but after som testing, i finally got it working by just copy the file `2` to `.config/fastfetch` folder, rename that, and add all my configs/lines to it. 

I don't know why, but to create a new file just wont work. Might be a bug? 

Anywho, here is my result (same but still a proof), thanks for all your help @CarterLi with my previous issue. 

![Screenshot_20230220-231827.png](https://user-images.githubusercontent.com/4266424/220207199-011ccda2-0112-4408-b35e-a6d3da568978.png)

**Reply by CarterLi** [reply]
May you create a issue with minimal reproduction steps provided?

**Reply by Dan1jel** [reply]
Will try to get this done asap :)
